src/scripts/main.js

```diff
@@ -1,3 +1,244 @@
 'use strict';
 
-// write your code here
+const startBtn = document.querySelector('.button');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+const score = document.querySelector('.game-score');
+const gameScore = document.querySelector('.game-score');
+const rows = document.querySelectorAll('.field-row');
+
+let arrOfNumbers = [
+  ['', '', '', ''],
+  ['', '', '', ''],
+  ['', '', '', ''],
+  ['', '', '', ''],
+];
+let moves = 0;
+let maxNum = 0;
+
+score.innerText = maxNum;
+
+startBtn.addEventListener('click', (e) => {
+  e.target.classList.toggle('restart');
+
+  if (e.target.textContent === 'Start') {
+    e.target.textContent = 'Restart';
+    maxNum = 0;
+
+    arrOfNumbers = [
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+    ];
+    randomNum();
+    paint();
+  } else {
+    e.target.textContent = 'Start';
+  }
+});
+
+if (maxNum >= 2048) {
+  messageWin.classList.toggle('hidden');
+  messageStart.classList.toggle('hidden');
+}
+
+function hasEmpty() {
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      if (arrOfNumbers[i][j] === '') {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function maxValue() {
+  for (const item of arrOfNumbers) {
+    for (const num of item) {
+      if (num !== '' && maxNum < num) {
+        maxNum = num;
+        messageStart.classList.toggle('hidden');
+      }
+    }
+  }
+
+  gameScore.innerText = maxNum;
+}
+
+function randomNum() {
+  if (!hasEmpty()) {
+    messageLose.classList.toggle('hidden');
+    messageStart.classList.toggle('hidden');
+
+    return;
+  }
+
+  let x;
+  let y;
+  const value = 2;
```

Comment:
The variable 'value' is always 2. According to task requirements, after each move, a 2 or 4 should appear in a random empty cell, with 4 having a 10% probability.